### PR TITLE
Hide other player's privates

### PR DESCRIPTION
**Pull request by ventusignis**
Implementation of https://github.com/tobymao/18xx/issues/1153

### CODE REVIEW

**Review by jenf — APPROVED** [review]

File: assets/app/view/game/buy_companies.rb [hunk 1]
```diff
@@ -32,11 +33,34 @@ def render_companies
           [company.owner == @corporation.owner ? 0 : 1, company.value]
         end
 
-        companies.map do |company|
+        companies_to_buy = companies.map do |company|
+          unless company.owner == @corporation.owner || @show_other_players
```
Comment (tobymao) [inline assets/app/view/game/buy_companies.rb:37]: don't use unless with multiple conditions

File: assets/app/view/game/buy_companies.rb [hunk 2]
```diff
@@ -32,11 +33,34 @@ def render_companies
...
+        if hidden_companies
+          companies_to_buy << h('button.no_margin',
+                                { on: { click: -> { store(:show_other_players, true) } }, **button_props },
+                                'Show companies from other players')
+        elsif @show_other_players
+          companies_to_buy << h('button.no_margin',
+                                { on: { click: -> { store(:show_other_players, false) } }, **button_props },
+                                'Hide companies from other players')
+        end
+        companies_to_buy
```
Comment (tobymao) [inline assets/app/view/game/buy_companies.rb:63]: this needs a compact  because you're adding nils now on 39

**Comment by ventusignis** [comment]
image
image

File: assets/app/view/game/buy_companies.rb (hunk 1 above)
Comment (ventusignis) [reply assets/app/view/game/buy_companies.rb]: I've replaced this with the equivalent `if` statement

File: assets/app/view/game/buy_companies.rb (hunk 2 above)
Comment (ventusignis) [reply assets/app/view/game/buy_companies.rb]: Fixed, thanks!

**Pull request merged**